gcloud-java-pubsub/src/main/java/com/google/cloud/pubsub/TopicInfo.java

```diff
@@ -98,15 +98,16 @@ public int hashCode() {
     return Objects.hash(name);
   }
 
+  final boolean baseEquals(TopicInfo topicInfo) {
+    return Objects.equals(name, topicInfo.name);
+  }
+
   @Override
   public boolean equals(Object obj) {
-    if (this == obj) {
-      return true;
-    }
-    if (obj == null || !obj.getClass().equals(this.getClass())) {
-      return false;
-    }
-    return Objects.equals(name, ((TopicInfo) obj).name);
+    return obj == this
```

Comment:
The reason I changed is just for consistency with other classes (see [BlobInfo](https://github.com/GoogleCloudPlatform/gcloud-java/blob/master/gcloud-java-storage/src/main/java/com/google/cloud/storage/BlobInfo.java#L640) or [ZoneInfo](https://github.com/GoogleCloudPlatform/gcloud-java/blob/master/gcloud-java-dns/src/main/java/com/google/cloud/dns/ZoneInfo.java#L295)). If you have a preference for conditionals I don't mind changing though we should change it everywhere to be consistent.